examples/07_end_to_end/seismic_waveform.ipynb

```diff
@@ -0,0 +1,1322 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f375a246-44ee-474b-af5b-f56ee3a6fda8",
+   "metadata": {},
+   "source": [
+    "# Processing Seismic Waveform Data with `obspy` and `csp`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "b25fa8e0-ba15-4f22-b50d-3c50f8c909ed",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import csp\n",
+    "import logging\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import sys\n",
+    "import warnings\n",
+    "from csp import ts\n",
+    "from datetime import timedelta, datetime\n",
+    "from obspy import read, Trace, Stream, UTCDateTime\n",
+    "from scipy.signal import iirfilter, sosfilt, zpk2sos\n",
+    "from typing import Dict, List, Optional, Tuple\n",
+    "\n",
+    "warnings.filterwarnings('ignore') "
+   ]
```

Comment:
I'd put this code after the introduction and make a note that the user must `pip install obspy` as it's not a required csp dependency. Additionally, `scipy` isn't a required csp dependency either.